source/common/router/config_impl.h

```diff
@@ -487,7 +488,7 @@ class RouteEntryImplBase : public RouteEntry,
   const std::multimap<std::string, std::string> opaque_config_;
 
   const DecoratorConstPtr decorator_;
-  const Http::Code redirect_response_code_;
+  const Http::Code direct_response_code_;
```

Comment:
nit: I might consider making this `Optional<Http::Code>`. I think it would be slightly clearer in all the places you set and consume it.